libs/ui/src/lib/modal/modal.tsx

```diff
@@ -0,0 +1,16 @@
+import clsx from 'clsx';
+import { useMemo } from 'react';
+
+export interface IModal {
+  open: boolean;
+  children: React.ReactNode;
+}
+
+export function Modal({ open, children }: IModal) {
+  const modalClass = useMemo(
+    () =>
+      clsx('modal bg-black/50', { 'modal-open': open, 'modal-closed': !open }),
+    [open],
+  );
+  return <div className={modalClass}>{children}</div>;
```

Comment:
Might be worth adding a close click area here. That way clicking outside the content will close the window automatically.

```suggestion
  const [isOpen, setIsOpen] = useState(open);
  
  useEffect(() => {
    setIsOpen(open);
  }, [open]);

  return <div className={modalClass}>
    <div onClick={() => setIsOpen(false)} className="absolute top-0 left-0 w-full h-full" />
    {children}
  </div>;
```

Above suggestion should give you two way binding control. If that gives you any trouble with your current implementation it's okay to skip on this for now.

Reply:
I was wondering how to do this. This works great! But we just need to coordinate the z-index we applied to different layers in the future, so far
xterm: z-10
close area: z-20
command pallet: z-50

Reply:
Yeah the parent class will probably need a Z index. We should probably figure out some sorta strategy to our z index layers 🤔 Maybe Ben has a good idea on how to best organize this as Tailwind's z index layers are pretty vague.